### PR TITLE
chore: remove gson

### PR DESCRIPTION
**Pull request by anonvt**
Removes Gson in favor of manually parsing with JSONObject

Closes: https://topsort.atlassian.net/browse/API-854

### CODE REVIEW

File: TopsortAnalytics/src/main/java/com/topsort/analytics/model/ClickEvent.kt [hunk 1]
```diff
@@ -41,7 +42,32 @@ data class Click(
...
+) {
+    fun toJsonObject(): JSONObject {
+        return JSONObject()
+            .put("resolvedBidId", resolvedBidId)
+            .put("entity", entity?.toJsonObject())
+            .put("additionalAttribution", additionalAttribution)
+            .put("placement", placement.toJsonObject())
+            .put("occurredAt", occurredAt)
+            .put("opaqueUserId", opaqueUserId)
+            .put("id", id)
```
Comment (PabloReszczynski) [inline TopsortAnalytics/src/main/java/com/topsort/analytics/model/ClickEvent.kt]: With this you may end up with an object that has both a `resolvedBidId` and an `entity` which is a bad request according to our API.
The best solution is to ask if whether it has a `resolvedBidId`, then use that, otherwise, use the entity.

File: TopsortAnalytics/src/main/java/com/topsort/analytics/model/ClickEvent.kt [hunk 2]
```diff
@@ -41,7 +42,32 @@ data class Click(
...
+                entity = Entity.fromJsonObject(json.getJSONObject("entity")),
+                additionalAttribution = json.getString("additionalAttribution"),
+                placement = Placement.fromJsonObject(json.getJSONObject("placement")),
+                occurredAt = json.getString("occurredAt"),
+                opaqueUserId = json.getString("opaqueUserId"),
+                id = json.getString("id"),
+            )
+        }
+    }
+}
```
Comment (PabloReszczynski) [inline TopsortAnalytics/src/main/java/com/topsort/analytics/model/ClickEvent.kt]: On which situations would you want to parse a json event? I think that, other than for testing purposes, this is not needed.

**Review by PabloReszczynski — CHANGES_REQUESTED** [review]
Some comments. Other than that, it looks good to me.

**Review by PabloReszczynski — APPROVED** [review]

File: TopsortAnalytics/build.gradle [hunk 3]
```diff
@@ -41,13 +41,18 @@ dependencies {
     implementation 'androidx.appcompat:appcompat:1.6.1'
     implementation 'com.google.android.material:material:1.9.0'
 
+    //Gson
+    implementation 'com.google.code.gson:gson:2.9.0'
+
```
Comment (PabloReszczynski) [inline TopsortAnalytics/build.gradle]: Why did you add gson back?

Comment (anonvt) [reply TopsortAnalytics/build.gradle]: Just comparing results, and it won't compile/test without it while I didn't change the Cache class
It will be removed 🔪

File: TopsortAnalytics/src/main/java/com/topsort/analytics/model/ClickEvent.kt (hunk 1 above)
Comment (anonvt) [reply TopsortAnalytics/src/main/java/com/topsort/analytics/model/ClickEvent.kt]: Changed it, also changed the constructor to private and added Factories to Click and Impressions to discipline these objects creation.

File: TopsortAnalytics/src/main/java/com/topsort/analytics/model/ClickEvent.kt (hunk 2 above)
Comment (anonvt) [reply TopsortAnalytics/src/main/java/com/topsort/analytics/model/ClickEvent.kt]: Current cache implementation writes the serialized json to SharedPreferences (and re-reads it before send the event), I'll move this into test-specific classes when convenient.

**Review by PabloReszczynski — APPROVED** [review]

**Review by jbergstroem — APPROVED** [review]
